TLM/TLM/Manager/Impl/SpeedLimitManager.cs

```diff
@@ -512,7 +510,11 @@ public class SpeedLimitManager
                 NetInfo.Lane laneInfo = segmentInfo.m_lanes[laneIndex];
                 NetInfo.Direction d = laneInfo.m_finalDirection;
 
-                if (d == finalDir && laneInfo.MayHaveCustomSpeedLimits()) {
+                bool applicableDir = finalDir.HasValue
+                    ? (d == finalDir.Value)
+                    : (d == NetInfo.Direction.None);
```

Comment:
`d == None` or `d != None`?

Reply:
yup, that was one of the issues. also I'm adding in check for lane count (somewhat crufty using `SortedLanes` for now but still cleaner than what it was before.